base/Base_compiler.jl

```diff
@@ -288,7 +293,11 @@ process_sysimg_args!()
 
 function isready end
 
-include(strcat(BUILDROOT, "../usr/share/julia/Compiler/src/Compiler.jl"))
+if file_exists(strcat(BUILDROOT, "../usr/share/julia/Compiler/src/Compiler.jl"))
```

Comment:
I don't understand why the `usr/share`, resp `share` portion of this can't be folded into the definition of `BUILDROOT`. We're fully in control of that path both in the Makefile and in PackageCompiler. That seems like the right place for this complexity.

Reply:
Well, there's two options I think. One is to DATAROOTDIR that instead of BUILDROOT, since currently, I believe we always link `DATAROOTDIR/julia/base` to `BUILDROOT/base`, so that was what I meant above. However, on second thought, I'm not convinced that that's right at all. Various downstream packages expect the full base sources to be available in `usr/share`, which I imagine is broken right now for an out of tree build. That's out of the scope of this PR, but if you want to use two arguments to be robust for potential rearrangement there, I think that's fine.

Reply:
With the latest commit I can build a sysimage on both a release and source with https://github.com/JuliaLang/PackageCompiler.jl/pull/1002